### PR TITLE
Add module for all components

### PR DESCRIPTION
**Pull request by scote**
- Add lazy load to demo app page
- Add index.ts to folders.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -13,13 +13,12 @@ platform:
 
 install:
   - ps: Install-Product node $env:nodejs_version
-  - yarn install --frozen-lockfile --no-progress
+  - yarn install --frozen-locykfile --no-progress
```
Comment (jfcere) [inline appveyor.yml]: --frozen-lockfile (no y)

File: src/app/dropdown/dropdown.module.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzDropdownDividerComponent } from './dropdown-divider/dropdown-divider.component';
+import { MzDropdownItemComponent } from './dropdown-item/dropdown-item.component';
```
Comment (jfcere) [inline src/app/dropdown/dropdown.module.ts]:
```typescript
import { MzDropdownItemComponent } from './dropdown-item/';
```

File: src/app/dropdown/dropdown.module.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzDropdownDividerComponent } from './dropdown-divider/dropdown-divider.component';
```
Comment (jfcere) [inline src/app/dropdown/dropdown.module.ts]:
```typescript
import { MzDropdownDividerComponent } from './dropdown-divider';
```

File: src/app/icon/icon.module.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  declarations: [
+    MzIconMdiDirective,
+    MzIconDirective,
+  ],
+  exports: [
+    MzIconMdiDirective,
+    MzIconDirective,
+  ],
+})
+export class MzIconModule { }
```
Comment (jfcere) [inline src/app/icon/icon.module.ts]: Should we do 2 modules for icons or would it be too confusing?
- MzIconModule
- MzIconMdiModule

File: src/app/input/input.module.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzInputContainerComponent } from './input-container/input-container.component';
```
Comment (jfcere) [inline src/app/input/input.module.ts]:
```typescript
import { MzInputContainerComponent } from './input-container';
```

File: src/app/input/input.module.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { MzInputPrefixDirective } from './input-prefix/input-prefix.directive';
```
Comment (jfcere) [inline src/app/input/input.module.ts]:
```typescript
import { MzInputPrefixDirective } from './input-prefix';
```

File: src/app/modal/modal.module.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzModalCloseDirective } from './modal-close/modal-close.directive';
```
Comment (jfcere) [inline src/app/modal/modal.module.ts]:
```typescript
import { MzModalCloseDirective } from './modal-close';
```

File: src/app/modal/modal.module.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import { MzModalComponent, MzModalContentDirective, MzModalFooterDirective, MzModalHeaderDirective } from './modal.component';
+import { MzModalService } from './services/modal.service';
```
Comment (jfcere) [inline src/app/modal/modal.module.ts]:
```typescript
import { MzModalService } from './services';
```

File: src/app/navbar/navbar.module.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzNavbarItemContainerComponent } from './navbar-item-container/navbar-item-container.component';
```
Comment (jfcere) [inline src/app/navbar/navbar.module.ts]:
```typescript
import { MzNavbarItemContainerComponent } from './navbar-item-container';
```

File: src/app/navbar/navbar.module.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import { MzNavbarItemComponent } from './navbar-item/navbar-item.component';
```
Comment (jfcere) [inline src/app/navbar/navbar.module.ts]:
```typescript
import { MzNavbarItemComponent } from './navbar-item';
```

File: src/app/radio-button/radio-button.module.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzRadioButtonContainerComponent } from './radio-button-container/radio-button-container.component';
```
Comment (jfcere) [inline src/app/radio-button/radio-button.module.ts]:
```typescript
import { MzRadioButtonContainerComponent } from './radio-button-container';
```

File: src/app/select/select.module.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzSelectContainerComponent } from './select-container/select-container.component';
```
Comment (jfcere) [inline src/app/select/select.module.ts]:
```typescript
import { MzSelectContainerComponent } from './select-container';
```

File: src/app/sidenav/sidenav.module.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzSidenavCollapsibleHeaderComponent } from './sidenav-collapsible/sidenav-collapsible-header/sidenav-collapsible-header.component';
+import { MzSidenavCollapsibleComponent, MzSidenavCollapsibleContentDirective } from './sidenav-collapsible/sidenav-collapsible.component';
+import { MzSidenavDividerComponent } from './sidenav-divider/sidenav-divider.component';
+import { MzSidenavHeaderComponent } from './sidenav-header/sidenav-header.component';
+import { MzSidenavLinkComponent } from './sidenav-link/sidenav-link.component';
+import { MzSidenavSubheaderComponent } from './sidenav-subheader/sidenav-subheader.component';
```
Comment (jfcere) [inline src/app/sidenav/sidenav.module.ts]: all those imports can use chaining ...

```typescript
import {
  MzSidenavCollapsibleComponent,
  MzSidenavCollapsibleContentDirective,
  MzSidenavCollapsibleHeaderComponent,
} from './sidenav-collapsible';
import { MzSidenavDividerComponent } from './sidenav-divider';
import { MzSidenavHeaderComponent } from './sidenav-header';
import { MzSidenavLinkComponent } from './sidenav-link';
import { MzSidenavSubheaderComponent } from './sidenav-subheader';
```

File: src/app/switch/switch.module.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzSwitchContainerComponent } from './switch-container/switch-container.component';
```
Comment (jfcere) [inline src/app/switch/switch.module.ts]:
```typescript
import { MzSwitchContainerComponent } from './switch-container';
```

File: src/app/tab/tab.module.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzTabItemComponent } from './tab-item/tab-item.component';
```
Comment (jfcere) [inline src/app/tab/tab.module.ts]:
```typescript
import { MzTabItemComponent } from './tab-item';
```

File: src/app/textarea/textarea.module.ts [hunk 16]
```diff
@@ -0,0 +1,20 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzTextareaContainerComponent } from './textarea-container/textarea-container.component';
```
Comment (jfcere) [inline src/app/textarea/textarea.module.ts]:
```typescript
import { MzTextareaContainerComponent } from './textarea-container';
```

File: src/app/textarea/textarea.module.ts [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+import { MzTextareaPrefixDirective } from './textarea-prefix/textarea-prefix.directive';
```
Comment (jfcere) [inline src/app/textarea/textarea.module.ts]:
```typescript
import { MzTextareaPrefixDirective } from './textarea-prefix/';
```

File: src/app/timepicker/timepicker.module.ts [hunk 18]
```diff
@@ -0,0 +1,17 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzTimepickerContainerComponent } from './timepicker-container/timepicker-container.component';
```
Comment (jfcere) [inline src/app/timepicker/timepicker.module.ts]:
```typescript
import { MzTimepickerContainerComponent } from './timepicker-container';
```

File: src/app/toast/toast.module.ts [hunk 19]
```diff
@@ -0,0 +1,9 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzToastService } from './services/toast.service';
```
Comment (jfcere) [inline src/app/toast/toast.module.ts]:
```typescript
import { MzToastService } from './services';
```

File: src/app/validation/validation.module.ts [hunk 20]
```diff
@@ -0,0 +1,19 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { MzErrorMessageComponent } from './error-message/error-message.component';
```
Comment (jfcere) [inline src/app/validation/validation.module.ts]:
```typescript
import { MzErrorMessageComponent } from './error-message';
```

File: src/app/icon/icon.module.ts (hunk 4 above)
Comment (scote) [reply src/app/icon/icon.module.ts]: You are right we should have one module for each one.

**Comment by jfcere** [comment]
Still for this to be complete it needs ...
- Module name should be specified on each demo-page
- Add example of "shared module" like [Angular Material](https://material.angular.io/guide/getting-started#step-3-import-the-component-modules) in `readme.md`
- As it is not removed, `MaterializeModule` should still be mentioned in `readme.md` as an alternative to import everything

File: README.md [hunk 21]
```diff
@@ -142,20 +142,43 @@ You must import component module you want to use inside your module to be able t
...
++   MzInputModule,
   ],
   declarations: [ HomeComponent ],
 })
 export class HomeModule { }
 ```
 
+
+### Alternative
+As an alternative, you can import `MaterializeModule` to make all component availables. This module is **depracated** and it will be removed in a near futur.
```
Comment (jfcere) [inline README.md]: ~futur~ future

Comment (scote) [reply README.md]: I though i did this correction haha sorry

Comment (jfcere) [reply README.md]: lol no problem ;)  This is starting to looking good ... should be all set for merge.

**Pull request merged**